website_attribute_set/models/mixins.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2011 Kencove (http://www.kencove.com).
+# @author Mohamed Alkobrosli <malkobrosly@kencove.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+from difflib import SequenceMatcher
+
+from odoo import api, models
+from odoo.osv import expression
+
+_logger = logging.getLogger(__name__)
+
+
+def search_extra(env, search_term):
+    extra_domains = []
+    attributes = (
+        env["attribute.attribute"].sudo().search([("e_com_visibility", "=", True)])
+    )
+    for attribute in attributes:
+        if attribute.attribute_type in ["char", "text"]:
+            extra_domain = [(attribute.name, "ilike", search_term)]
+            extra_domains.append(extra_domain)
+        elif attribute.attribute_type in ["integer", "float"]:
+            try:
+                if attribute.attribute_type == "integer":
+                    extra_domain = [(attribute.name, "ilike", int(search_term))]
+                    extra_domains.append(extra_domain)
+                elif attribute.attribute_type == "float":
+                    extra_domain = [(attribute.name, "ilike", float(search_term))]
+                    extra_domains.append(extra_domain)
+            except ValueError as e:
+                _logger.info(f"{e}")
+        elif attribute.relation_model_id and attribute.attribute_type in [
+            "select",
+            "multiselect",
+        ]:
+            extra_domain = [(f"{attribute.name}.name", "ilike", search_term)]
+            extra_domains.append(extra_domain)
+        else:
+            similarity = (
+                SequenceMatcher(None, attribute.field_description, search_term).ratio()
+                * 100
+            )
+            if similarity > 80:
```

Comment:
```suggestion
            if similarity > 80:
```
What do we base this number (`80`) on and should we hard code it?

Reply:
it is a percentage of similarity like 80% alike